### DISCUSSION TITLE
Telugu Language Leader

### DISCUSSION THREAD

**Discussion by rPonuganti** [OHF-Voice/intents#419] [Language Leaders]
Telugu is a South Indian regional language. 
I would like to bring Telugu voice support to Home Assistant. I would be happy to be a leader for the Telugu language.
Telugu is my mother tongue. I can fluently speak English. Also German at moderate level.

**Comment by synesthesiam** [comment]
Thanks @rPonuganti! I will create some empty Telugu (`te`) templates so you can get started :+1: 

I'd be curious to hear what you think about the Telugu voices I trained here too: https://mycroftai.github.io/mimic3-voices/#te_IN_cmu-indic_low

**Reply by rPonuganti** [reply]
Hi @synesthesiam Surprised to learn that you trained Telugu voices  :-) 
The voices are good but a bit of robotic, few pronunciations are very robotic and . SS is too fast, KPN is too slow. SK sounds optimum in speed.

**Reply by synesthesiam** [reply]
Thanks for the feedback! I don't have a lot of Telugu data, but I'll train any voice I can find an open dataset for :slightly_smiling_face:

**Comment by synesthesiam** [comment]
@rPonuganti I'll watch for your PRs and add you as language leader once we get 2 small ones in :+1:

**Reply by rPonuganti** [reply]
@synesthesiam  I have a question regarding the Language Script.
Did I understand correctly that intents should be written in the Languages own script ? example like "తెలుగు"
But how does the machine understand this script ? Will the machine also be trained with the language specific scripts ?
Slots for devices and areas can still be English ??
In modern world unfortunately Telugu is Tenglish (Telugu + English)

**Reply by synesthesiam** [reply]
The intent matcher is not very smart, it will just match text without understanding it. You can mix scripts (Telugu + English) if that is how it would be recognized by a speech to text system like Google or Alexa.

For example, a sentence template could be "I speak తెలుగు" which would only match the 1 sentence. We're only focused on things like turning lights on and off, so we don't have to understand a lot :slightly_smiling_face:

**Reply by rPonuganti** [reply]
Here I go with the first PR #588 .

**Comment by rPonuganti** [comment]
@synesthesiam I got a new PR [#1348 ](https://github.com/home-assistant/intents/pull/1348) since there is no language leader it was not approved. 
Can you please approve it ?

Thanks.

**Reply by synesthesiam** [reply]
Sorry for the delay! Let me know if you'd be interested in being a Telugu language leader.

**Comment by rPonuganti** [comment]
Hi,

Yes sure I would be interested to be language leader for te.

Thanks.

On Thu 22. Jun 2023 at 03:03, Michael Hansen ***@***.***>
wrote:

> Sorry for the delay! Let me know if you'd be interested in being a Telugu
> language leader.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/intents/discussions/419#discussioncomment-6246226>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AASILOH7VD7ZGRPYUXXOTHDXMNSENANCNFSM6AAAAAATSO4NXM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Thanks & regards,
Raghavendra Ponuganti

**Comment by rPonuganti** [comment]
Hi, Telugu Language leader is still open. Can you make me the leader.?

 Thanks.

**Comment by KSruthiVel** [comment]
Hi @synesthesiam,

I came across the discussion on Telugu language support for Home Assistant and would like to contribute. I am also interested in taking on the role of Telugu language leader. I speak the Andhra dialect and can assist with translations, intent development, and improvements.

Thank you,
Krishna
